photon-client/src/main/scala/com/linkedin/photon/ml/cli/game/training/GameTrainingDriver.scala

```diff
@@ -637,21 +635,34 @@ object GameTrainingDriver extends GameDriver {
         // TODO: Match on this to make it clearer
         val evaluator = models.head._2.get.head._1
         val baseConfig = models.head._3
-        val evaluationFunction = new GameEstimatorEvaluationFunction(estimator, baseConfig, trainingData, testData)
 
         val range = getOrDefault(hyperParameterTuningRange)
-        val ranges = List.fill(evaluationFunction.numParams)(DoubleRange(log(range.start), log(range.end)))
+        val rangeWrapped = DoubleRange(log(range.start), log(range.end))
+
+        val evaluationFunction = new GameEstimatorEvaluationFunction(estimator, baseConfig, trainingData, testData, range)
+        val dimension = evaluationFunction.numParams
 
         val searcher = getOrDefault(hyperParameterTuning) match {
           case HyperparameterTuningMode.BAYESIAN =>
-            new GaussianProcessSearch[GameEstimator.GameResult](ranges, evaluationFunction, evaluator)
+            new GaussianProcessSearch[GameEstimator.GameResult](dimension, evaluationFunction, evaluator)
 
           case HyperparameterTuningMode.RANDOM =>
-            new RandomSearch[GameEstimator.GameResult](ranges, evaluationFunction)
+            new RandomSearch[GameEstimator.GameResult](dimension, evaluationFunction)
         }
         val observations = searcher.convertObservations(models)
```

Comment:
@basukinjal @joshvfleming I think here was a bug. In the previous version, `models` had values in [0.001, 1000], and observations had [-3, 3], because `convertObservations` did a wrapping. But we should keep all the prior data in [0.001, 1000]. So I suggest we move wrapping/unwrapping out of mapToVec/vecToMap and implement it separately.